lib/amazonka-core/src/Amazonka/Sign/V4.hs

```diff
@@ -42,7 +43,16 @@ presign ex rq a r ts = Base.signRequest meta mempty auth
         . pair (CI.original hAMZSignedHeaders) (toBS shs)
         . pair (CI.original hAMZToken) (toBS <$> _authSessionToken a)
 
-    digest = Base.Tag "UNSIGNED-PAYLOAD"
+    digest =
+      case _requestBody rq of
+        Chunked _ -> unsignedPayload
+        Hashed HashedStream {} -> unsignedPayload
+        Hashed (HashedBytes _ b) ->
+          if BS.length b == 0 && (_serviceSigningName . _requestService $ rq) == "s3"
+            then unsignedPayload
+            else Base.Tag . Chunked.sha256 $ b
+
+    unsignedPayload = Base.Tag "UNSIGNED-PAYLOAD"
```

Comment:
Please add a link to https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html indicating what this is and why.

Reply:
Done. See the haddock of the function

Reply:
I agree with putting it in the haddock - end users should be able to know what's going on.